library/compiler-builtins/compiler-builtins/src/aarch64_linux.rs

```diff
@@ -126,6 +135,41 @@ macro_rules! stxp {
     };
 }
 
+// If supported, perform the requested LSE op and return, or fallthrough.
+macro_rules! try_lse_op {
+    ($op: literal, $ordering:ident, $bytes:tt, $($reg:literal,)* [ $mem:ident ] ) => {
+        concat!(
+            ".arch_extension lse; ",
```

Comment:
AFAIK, there is currently no reliable way to use .arch_extension in naked_asm/global_asm correctly in general cases due to the LLVM bug. (See https://github.com/rust-lang/rust/pull/137720#discussion_r2014505753 and https://github.com/rust-lang/rust/pull/137720#discussion_r1973608259 for more info.) 
Although maybe there is some special rule for compiler-builtins...

Reply:
From experimentation, it seems like `.arch_extension` only applied to the function. Specifying the raw instruction as `.inst 0x...` without toggling features also works, but is ugly. If that's preferable, I'll update the PR.

Reply:
In the context of compiler-builtins, just using `.arch_extension` is probably fine since it is always compiled separately and is never included in LTO.